src/script/builder/tir/var.h

```diff
@@ -26,16 +26,36 @@ namespace script {
 namespace builder {
 namespace tir {
 
-tvm::tir::Buffer Buffer(Array<PrimExpr> shape,                 //
-                        DataType dtype = DataType::Float(32),  //
-                        String name = "buffer",                //
-                        String storage_scope = "");
-
-tvm::tir::Buffer DeclBuffer(Array<PrimExpr> shape, DataType dtype, String buffer_name,
-                            Optional<tvm::tir::Var> data, Array<PrimExpr> strides,
-                            PrimExpr elem_offset, String storage_scope, int align,
-                            int offset_factor, String buffer_type_str,
-                            Array<IntImm> axis_separators);
+class BufferNode : public runtime::Object {
+ public:
+  tvm::tir::Buffer buffer;
+
+  void VisitAttrs(tvm::AttrVisitor* v) { v->Visit("buffer", &buffer); }
+
+  static constexpr const char* _type_key = "script.builder.tir.Buffer";
+  TVM_DECLARE_BASE_OBJECT_INFO(BufferNode, runtime::Object);
+
+ public:
+  tvm::tir::BufferLoad BufferLoad(Array<PrimExpr> indices);
+  tvm::tir::BufferStore BufferStore(PrimExpr value, Array<PrimExpr> indices);
+  tvm::tir::BufferRegion BufferRegion(Array<Range> region);
+  tvm::tir::Prefetch Prefetch(Array<Range> bounds);
```

Comment:
in C++, do we want to enable overloaded `operator[]` that dispatches to `BufferLoad` and `BufferRegion`?

Reply:
also, given these member methods are already implemented outside, probably we dont need those member methods?